spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/EurekaClientAutoConfiguration.java

```diff
@@ -91,6 +91,9 @@
 	@Value("${eureka.instance.hostname:${EUREKA_INSTANCE_HOSTNAME:}}")
 	private String hostname;
 
+	@Value("${eureka.instance.prefer-ip-address:${eureka.instance.preferIpAddress:false}}")
```

Comment:
This should be moved to use a `RelaxedPropertyReslover` in the `@Bean` method itself. The resolver from https://github.com/windinn/spring-cloud-netflix/blob/4f801d8653ca8b178a4b8cb1740736a4b0dfc435/spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/EurekaClientAutoConfiguration.java#L132-L134 should be moved to the beginning of the method. The `hostname` field should be moved also. Without it, you wouldn't be able to use an ENV var like `EUREKA_INSTANCE_PREFER_IP_ADDRESS`.